intermediate_notebooks/E2E/taxi/NYCTaxi-E2E-GPU-Dask.ipynb

```diff
@@ -0,0 +1,1904 @@
+   "source": [
+    "pred.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "true.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate RMSE\n",
+    "squared_error = ((pred-true)**2)\n",
+    "\n",
+    "# compute the actual RMSE over the full test set\n",
+    "cupy.sqrt(squared_error.mean().compute())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save Trained Model for Later Use¶"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To make a model maximally useful, you need to be able to save it for later use. We'll use Google Cloud Storage to persist the trained model in a dill file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gcsfs, dill\n",
+    "\n",
+    "fs = gcsfs.GCSFileSystem()\n",
+    "# replace with a bucket you own\n",
+    "bucket = 'rapidsai-test-1/'\n",
+    "\n",
+    "with fs.open(bucket+'trained_model.dill', 'wb') as file:  \n",
+    "    dill.dump(trained_model, file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As an alternative, you can save the trained model on your system as below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Save the model to file\n",
+    "booster.save_model('xgboost-model')\n",
+    "print('Training evaluation history:', history)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reload a Saved Model from Disk"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also read the saved model back out of Google Cloud Storage and into a normal XGBoost model object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with fs.open(bucket+'trained_model.dill', 'rb') as file:  \n",
+    "    model_from_disk = dill.load(file)\n",
+    "\n",
+    "# Generate predictions on the test set again, but this time using the reloaded model\n",
+    "prediction = xgb.dask.predict(client, model_from_disk, X_test).persist()\n",
+    "wait(prediction)\n",
+    "\n",
+    "# Verify that the predictions result in the same RMSE error\n",
+    "pred = pred.map_partitions(lambda part: cudf.Series(part)).reset_index(drop=True)\n",
+    "true = Y_test['fare_amount'].reset_index(drop=True)\n",
+    "\n",
+    "squared_error = ((pred-true)**2)\n",
+    "\n",
+    "# compute the actual RMSE over the full test set\n",
+    "cupy.sqrt(squared_error.mean().compute())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## VISUALS with CUXFILTER"
```

Comment:
Now that we have the cuXfilter viz example [here](https://github.com/rapidsai/cuxfilter/blob/e78b148f6db9f1b0c2352d32255e7477decf5f21/notebooks/NYC_taxi_example.ipynb) let's remove this whole section.